iocore/net/SSLUtils.cc

```diff
@@ -1356,7 +1362,7 @@ SSLPrivateKeyHandler(SSL_CTX *ctx, const SSLConfigParams *params, const ats_scop
     return false;
   }
 
-  if (!SSL_CTX_check_private_key(ctx)) {
+  if (e == nullptr && !SSL_CTX_check_private_key(ctx)) {
```

Comment:
What cases do we need to check `e == nullptr`?
Are we really OK even if `SSL_CTX_check_private_key()` returned `false`, if `e` was not a `nullptr`?

Reply:
If there is no engine, then we definitely want to enforce that the private and public keys.

Depending on the engine, we may or may not be able to do the key check.  In the engine I am testing, the private key check is not possible at this point.  So requiring the check will cause my engine to fail to load.  

On other engines that rely on crypto hardware like TPM, the private key check will not work either.  

Perhaps eventually we can make it a setting knob, but for now only checking if we are not using an engine seemed like a reasonable tradeoff.

Reply:
I see. I guess that is the reason why there's no `return false` after the newly added `SSLError()`, right? I missed the difference when I saw it for the first time. It would be nice if there's a comment on the blocks.